src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2299,6 +2307,13 @@ cldnn::network::ptr primitive_inst::get_unfused_subgraph() {
                 data_prim.mem = data_node.get_attached_memory_ptr();
                 t.add(data_prim);
             } else {
+                cldnn::primitive_id dep_id = dep.first->id();
+                if (_node->is_type<fully_connected>() || _node->is_type<convolution>() || _node->is_type<deconvolution>()) {
+                    const auto& prim = _node->get_primitive();
+                    if (static_cast<size_t>(dep.second) == prim->input_size()) {
+                        dep_id = prim->dependencies()[prim->input_size()].pid;
+                    }
+                }
```

Comment:
Can we remove this change?

Reply:
Reverted, looks like https://github.com/openvinotoolkit/openvino/pull/26563 makes a similar fix

Reply:
Actually, my comment was only for this part not for other changes. The other changes should be remained as it was.
https://github.com/openvinotoolkit/openvino/pull/26563 will be updated not to touch the weight.